Polyhedron/demo/Polyhedron/MainWindow.cpp

```diff
@@ -1652,50 +1663,49 @@ void MainWindow::on_actionPreferences_triggered()
   connect(prefdiag.polyRadioButton, &QRadioButton::toggled,
           this, &MainWindow::set_facegraph_mode_adapter);
   
-  QStandardItemModel* iStandardModel = new QStandardItemModel(this);
-  //add blacklisted plugins
-  Q_FOREACH(QString name, plugin_blacklist)
-  {
-    QStandardItem* item =  new QStandardItem(name);
-    item->setCheckable(true);
-    item->setCheckState(Qt::Checked);
-    iStandardModel->appendRow(item);
-  }
+  //QStandardItemModel* iStandardModel = new QStandardItemModel(this);
 
-  //add operations plugins
-  Q_FOREACH(PluginNamePair pair,plugins){
-    QStandardItem* item =  new QStandardItem(pair.second);
-    item->setCheckable(true);
-    iStandardModel->appendRow(item);
-  }
-  
-  //add io-plugins
-  Q_FOREACH(CGAL::Three::Polyhedron_demo_io_plugin_interface* plugin, io_plugins)
-  {
-    QStandardItem* item =  new QStandardItem(plugin->name());
-    item->setCheckable(true);
-    if ( plugin_blacklist.contains(plugin->name()) ) item->setCheckState(Qt::Checked);
-    iStandardModel->appendRow(item);
-  }
+  std::vector<QTreeWidgetItem*> items;
 
-  //Setting the model
-  prefdiag.listView->setModel(iStandardModel);
-  
+  //add blacklisted plugins
+  Q_FOREACH (QDir dir, plugins_directories) {
+    QTreeWidgetItem* pluginItem = new QTreeWidgetItem(prefdiag.treeWidget);
+    pluginItem->setText(1, dir.absolutePath());
+    prefdiag.treeWidget->setItemExpanded(pluginItem, true);
+    QFont boldFont = pluginItem->font(1);
+    boldFont.setBold(true);
+    pluginItem->setFont(1, boldFont);
+    Q_FOREACH(QString fileName, dir.entryList(QDir::Files))
+    {
+      if(!fileName.contains("plugin") || !QLibrary::isLibrary(fileName))
+        continue;
+      QFileInfo fileinfo(fileName);
+      //set plugin name
+      QString name = fileinfo.fileName();
+      name.remove(QRegExp("^lib"));
+      name.remove(QRegExp("\\..*"));
+      QTreeWidgetItem *item = new QTreeWidgetItem(pluginItem);
+      item->setText(1, name);
+      if(plugin_blacklist.contains(name))
+        item->setCheckState(0, Qt::Checked);
+      else
+        item->setCheckState(0, Qt::Unchecked);
+      items.push_back(item);
+    }
+  }
```

Comment:
The `loadPlugins` function should store enough information, so that this one does not have to scan the directories again.

And, anyway, we do not care which are the available at the time the dialog is displayed: we want to know which ones were actually loaded.